modules/zone/variables.tf

```diff
@@ -16,7 +16,7 @@ variable "zone_ipv4_cidr" {
   type        = string
   default     = "10.128.0.0/16"
   validation {
-    condition = try(cidrnetmask(var.zone_ipv4_cidr), null) == "255.255.0.0" && !contains(cidrsubnets("172.16.0.0/15", 1, 1), var.zone_ipv4_cidr)
-    error_message = "CIDR for the zone network must be /16 and cannot overlap with 172.16.0.0/15"
+    condition = try(cidrnetmask(var.zone_ipv4_cidr), null) == "255.255.0.0" && !contains(cidrsubnets("10.0.0.0/8", 1, 1), var.zone_ipv4_cidr)
```

Comment:
The condition is wrong:
1. Need to remove the negation before the `contains()`. The condition must return true if input is valid.
2. The `cidrsubnets()` is wrong, this will generate ["10.0.0.0/9", "10.128.0.0/9"]. You'd need to write something like `cidrsubnets("10.0.0.0/8", 8, 8, 8, 8, [8 252 more times])`, though hopefully there is a better way! 
 
(Previously we checked that the subnet was _not_ one of 172.16.0.0/16 nor 172.17.0.0/16)